### ISSUE TITLE
Add Payment Method: Stripe processes even if the gateway is not selected.

### ISSUE THREAD

**Issue by msaggiorato** [woocommerce/woocommerce-gateway-stripe#553]
If you try to add a new card, and other gateway is selected, the stripe gateway is still processing code.

After doing a bit of debugging i think the error is on [this line](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/assets/js/stripe.js#L578), because you may need to the check isStripeChosen before the other checks, to do nothing if stripe is not checked.

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

**Comment by roykho** [comment]
Do you mean in My-Account: Add Payment Method page?

**Comment by msaggiorato** [comment]
Yes. In my debugging the first if fails correctly, but then it goes to [this if](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/assets/js/stripe.js#L662) and at that point, there's no check to see if we're actually trying to add a stripe payment.

**Comment by roykho** [comment]
Please see if this PR fixes it for you. https://github.com/woocommerce/woocommerce-gateway-stripe/pull/554

**Comment by msaggiorato** [comment]
Fixed for me. But shouldn't the check be done only once before the if structure? Since the check is done for both if statements.

**Comment by roykho** [comment]
sure works either way.
